### PR TITLE
November prototype - footer

### PR DESCRIPTION
**Pull request by helenb**
### What is the context of this PR?
New styling for the footer area for the November prototype.

### How to review
Preview any page in your local build to see the footer.

<details>
<summary>
Screenshots
</summary>

**Desktop:**
Screenshot 2024-10-09 at 12 20 13

**Mobile:**
Screenshot 2024-10-09 at 12 20 26

</details>

### CODE REVIEW

File: ons_alpha/jinja2/component_overrides/footer/_macro.njk
Comment (helenb) [inline ons_alpha/jinja2/component_overrides/footer/_macro.njk:1]: @ababic this overrides the design system template in order to add classes more granularly than the options allow, and in order to sort things like icons with hard-coded fills.

File: ons_alpha/static_src/sass/components/design_system_overrides/_container.scss
Comment (helenb) [inline ons_alpha/static_src/sass/components/design_system_overrides/_container.scss]: I realised the new design had a slightly different maximum width on all the container classes

Comment (helenb) [reply ons_alpha/static_src/sass/components/design_system_overrides/_container.scss]: Now tweaked to account for the padding

File: ons_alpha/jinja2/component_overrides/footer/_macro.njk
Comment (ababic) [inline ons_alpha/jinja2/component_overrides/footer/_macro.njk:37]: Is the `call` bit necessary on this line? Seems a bit out-of-place with all of the other component invocations

File: ons_alpha/jinja2/component_overrides/footer/_macro.njk
Comment (ababic) [inline ons_alpha/jinja2/component_overrides/footer/_macro.njk:54]: Is `params.button.classes` meant to be a list? If so, this might be better:

```suggestion
                        "classes": params.button.classes or [],
```

File: ons_alpha/jinja2/component_overrides/footer/_macro.njk
Comment (ababic) [inline ons_alpha/jinja2/component_overrides/footer/_macro.njk:165]: This code looks quite odd - is it jinja specific? Does the following not work?

```suggestion
                                                    "url": params.OGLLink.url 
or 'https://www.nationalarchives.gov.uk/doc/open-government-licence/version/3/',
                                                    "linkText": params.OGLLink.link or 'Drwydded Llywodraeth Leol v3.0',
```

**Review by ababic — COMMENTED** [review]
@helenb Looking generally fine, but had a few questions around some of the syntax. Maybe it's just my lack of familiarity with Jinja, but some things look a little odd.

File: ons_alpha/jinja2/component_overrides/footer/_macro.njk
Comment (helenb) [reply ons_alpha/jinja2/component_overrides/footer/_macro.njk:37]: As mentioned earlier in stand-up, this is basically a copy of the footer code in order to inject a few extra classes - so I don't really want to spend time fixing their code here.

File: ons_alpha/jinja2/component_overrides/footer/_macro.njk
Comment (helenb) [reply ons_alpha/jinja2/component_overrides/footer/_macro.njk:54]: Again, same comment that it's their code that is duplicated here so I can add extra classes.

File: ons_alpha/jinja2/component_overrides/footer/_macro.njk
Comment (helenb) [reply ons_alpha/jinja2/component_overrides/footer/_macro.njk:165]: And ditto...

**Comment by helenb** [comment]
@ababic would you be happy to approve it as is, given the footer macro code is copied from the design system?

**Review by ababic — APPROVED** [review]
@helenb If this is working for you locally, I see no need to hold up approval/deployment, so approving on that basis :)

**Pull request merged**
